Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -23,7 +23,6 @@ VertexShaderUid GetVertexShaderUid()
   memset(uid_data, 0, sizeof(*uid_data));
 
   _assert_(bpmem.genMode.numtexgens == xfmem.numTexGen.numTexGens);
-  _assert_(bpmem.genMode.numcolchans == xfmem.numChan.numColorChans);
```

Comment:
This has fixed the Chapter 6-2 crash and the shader effect seems to be visually identical to a real Wii.